mlflow/sklearn/utils.py

```diff
@@ -342,15 +465,17 @@ def _log_specialized_estimator_content(fitted_estimator, run_id, fit_args, fit_k
     import sklearn
 
     name_metric_dict = {}
+    artifact_dir = None
     try:
         if sklearn.base.is_classifier(fitted_estimator):
             name_metric_dict = _get_classifier_metrics(fitted_estimator, fit_args, fit_kwargs)
+            artifact_dir = _get_classifier_artifacts(fitted_estimator, fit_args, fit_kwargs)
```

Comment:
If we fail to log metrics, can't we still try to log artifacts & vice versa? It seems like these should be separate try/catch blocks.

Reply:
I create another separate try/catch block.